File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Money.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.NodeReference;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.NodeObjectBase;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.internal.OutputBuilderTypeChecker;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

@SuppressWarnings("MissingOverride")
public class Money extends ObjectBase implements SearchHit {

    public static final Type<Money> Reflection = Type.ofClass(Money.class);

    public static final class Fields implements TypeFields<Money> {
        private Fields() {}

        public static final Field<Money> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Money> amount =
                                Field.of("amount", Reflection);

                public static final Field<Money> currency =
                                Field.of("currency", Reflection);

    }

    public Money(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    @SuppressWarnings("UnusedMethod")
    private Money(InternalContext context, Map<String, Object> data) {
        super(context, data, "Money");
    }

    private Money(InternalContext context, ObjectBase base, Map<String, Object> data) {
        super(context, base, data, "Money");
    }

    public Money(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }
        public double getAmountOrThrow(String alias) {
            return fetchScalar("amount", alias);
        }

        public double getAmountOrThrow() {
            return fetchScalar("amount", null);
        }

        public Double getAmount(String alias) {
            return nullOnDataFailure(() -> fetchScalar("amount", alias));
        }

        public Double getAmount() {
            return nullOnDataFailure(() -> fetchScalar("amount", null));
        }

        public String getCurrencyOrThrow(String alias) {
            return fetchScalar("currency", alias);
        }

        public String getCurrencyOrThrow() {
            return fetchScalar("currency", null);
        }

        public String getCurrency(String alias) {
            return nullOnDataFailure(() -> fetchScalar("currency", alias));
        }

        public String getCurrency() {
            return nullOnDataFailure(() -> fetchScalar("currency", null));
        }


    public Builder toBuilder() {
        return new Builder(__context(), toBuilderBase());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context), null);
    }

    public static class Builder {
        private final InternalContext __context;
        private final ObjectBase __base;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context, ObjectBase __base) {
            this.__context = __context;
            this.__base = __base;
        }

                public Builder amount(double amount) {
                    amount = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Money",
                            "amount",
                            null,
                            amount);
                    data.put("amount", amount);
        return this;
                }

                public Builder currency(String currency) {
                    currency = OutputBuilderTypeChecker.checkField(
                            __context,
                            "Money",
                            "currency",
                            null,
                            currency);
                    data.put("currency", currency);
        return this;
                }


        public Money build() {
            return new Money(__context, __base, new LinkedHashMap<>(data));
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Money.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class)
class Money(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object,viaduct.api.grts.SearchHit
{
     fun getAmountOrThrow(alias: String?): kotlin.Double = TODO()
     fun getAmountOrThrow(): kotlin.Double = TODO()
     fun getAmount(alias: String?): kotlin.Double? = TODO()
     fun getAmount(): kotlin.Double? = TODO()

     fun getCurrencyOrThrow(alias: String?): kotlin.String = TODO()
     fun getCurrencyOrThrow(): kotlin.String = TODO()
     fun getCurrency(alias: String?): kotlin.String? = TODO()
     fun getCurrency(): kotlin.String? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Money =
            Builder(context).apply(block).build()
    }

    class Builder : ObjectBase.Builder<Money> {
        constructor(context: ExecutionContext)
            : super(
                context as InternalContext,
                TODO() as graphql.schema.GraphQLObjectType,
                null
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(context, type, baseEngineObjectData)

                  fun amount(value: kotlin.Double): Builder = TODO()

                  fun currency(value: kotlin.String): Builder = TODO()


        final override fun build(): Money = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Money> {
        override final val name = "Money"
        override final val kcls = viaduct.api.grts.Money::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Money> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Money> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Money.Reflection)

            final val amount: viaduct.api.reflect.Field<viaduct.api.grts.Money> =
                viaduct.api.internal.FieldImpl("amount", viaduct.api.grts.Money.Reflection)

            final val currency: viaduct.api.reflect.Field<viaduct.api.grts.Money> =
                viaduct.api.internal.FieldImpl("currency", viaduct.api.grts.Money.Reflection)

    }

}